src/map/tests.rs

```diff
@@ -593,6 +593,219 @@ fn iter_default() {
     assert_default::<IntoValues<K, V>>();
 }
 
+#[test]
+fn get_index_mut2() {
+    let mut map: IndexMap<i32, i32> = IndexMap::new();
+    map.insert(1, 2);
+    map.insert(3, 4);
+    map.insert(5, 6);
+
+    {
+        let (key, value) = map.get_index_mut2(0).unwrap();
+        assert_eq!(*key, 1);
+        assert_eq!(*value, 2);
+
+        *value = 7;
+    }
+    assert_eq!(map[0], 7);
+
+    {
+        let (key, _) = map.get_index_mut2(0).unwrap();
+        *key = 8;
+    }
+    assert_eq!(map.get_index(0).unwrap().0, &8);
+}
+
+#[test]
+fn shift_shift_remove_index() {
+    let mut map: IndexMap<i32, i32> = IndexMap::new();
+    map.insert(1, 2);
+    map.insert(3, 4);
+    map.insert(5, 6);
+    map.insert(7, 8);
+    map.insert(9, 10);
+
+    let result = map.shift_remove_index(1);
+    assert_eq!(result, Some((3, 4)));
+    assert_eq!(map.len(), 4);
+
+    map.swap_remove(&5);
```

Comment:
This swap is a bad choice, because it makes the other shifts irrelevant. That is, I can change all of those other shifts to swap (e.g. as if the shift was implemented incorrectly), and the expected values here still pass!

Just forget this line, and the rest could be more reasonable.

- `shift_remove_index(1) -> (3, 4)`
- `shift_remove_index(1) -> (5, 6)` -- because it **shifted** to fill the gap)
- `shift_remove_index(2) -> (9, 10)` -- because the last was never swapped, only shifted down twice, and it also tests that shift-removing the last entry is fine
- `shift_remove_index(2) -> None`